### PR TITLE
Add delete button and UnRecruit logic

### PR DESCRIPTION
**Pull request by qu-bit1**
- Deletes students from an event (Delete all & delete individually)
- Implemented UnRecruit logic in both the delete functions

### CODE REVIEW

**Comment by qu-bit1** [comment]
fixing linting issues

File: application/admin.student.go [hunk 1]
```diff
@@ -147,3 +147,56 @@ func postStudentsByEventHandler(mail_channel chan mail.Mail) gin.HandlerFunc {
...
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": "Invalid student ID"})
+		return
+	}
+
+	rcID, err := util.ParseUint(ctx.Param("rid"))
+	if err != nil {
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": "Invalid r ID"})
+		return
+	}
+	if err = rc.UnRecruitStudent(ctx, studentID, rcID); err != nil {
```
Comment (1-Harshit) [inline application/admin.student.go:169]: What?

**Review by 1-Harshit — CHANGES_REQUESTED** [review]
This is not right flow, try to make one api EXACTLY INVERSE of register student api. For consistency and readability.

File: application/admin.student.go (hunk 1 above)
Comment (1-Harshit) [inline application/admin.student.go:169]: this should only be invoked if the event is "Recruited" or "PPO" right?

Comment (qu-bit1) [reply application/admin.student.go:169]: yes fixing that and a few more things

**Pull request merged**